app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkTopicsCompletedFragmentTest.kt

```diff
@@ -327,31 +296,38 @@ class MarkTopicsCompletedFragmentTest {
       createMarkTopicsCompletedTestActivityIntent(internalProfileId)
     ).use {
       testCoroutineDispatchers.runCurrent()
-      scrollToPosition(position = 0)
       performItemCheckOnTopicSummaryListItem(itemPosition = 0)
-      scrollToPosition(position = 1)
       performItemCheckOnTopicSummaryListItem(itemPosition = 1)
-      scrollToPosition(position = 2)
       performItemCheckOnTopicSummaryListItem(itemPosition = 2)
-      scrollToPosition(position = 3)
       performItemCheckOnTopicSummaryListItem(itemPosition = 3)
       onView(withId(R.id.mark_topics_completed_all_check_box)).check(matches(isChecked()))
     }
   }
 
+  @Test
+  fun testMarkTopicsCompletedFragment_deselectAllTopics_deselectsAllTopics() {
+    launch<MarkTopicsCompletedTestActivity>(
+      createMarkTopicsCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.mark_topics_completed_all_check_box_container)).perform(click())
+      onView(withId(R.id.mark_topics_completed_all_check_box_container)).perform(click())
```

Comment:
For this & the other new tests: it would be nice to include a comment to make it clear that the two redundant lines are intentional. E.g. something like:

```suggestion
      // Click one to select all topics.
      onView(withId(R.id.mark_topics_completed_all_check_box_container)).perform(click())

      // Click a second time to unselect all topics.
      onView(withId(R.id.mark_topics_completed_all_check_box_container)).perform(click())

```

Reply:
Yeah @BenHenning for sure this makes sense, myself i had a hard time to understand the logic.